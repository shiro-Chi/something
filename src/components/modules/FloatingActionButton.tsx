import React from 'react';
import Fab from '@mui/material/Fab';
import AddIcon from '@mui/icons-material/Add';
import FormDialog from './FormDialog';
import { SxProps, Theme } from '@mui/material/styles';

/**
 * @interface FloatingActionButtonProps
 */
// eslint-disable-next-line @typescript-eslint/no-empty-interface
interface FloatingActionButtonProps {}

/** @type {SxProps} フローティングアクションボタンコンポーネントのスタイル定義 */
const style: SxProps<Theme> = {
  position: 'absolute',
  bottom: 20,
  right: 20,
};

/**
 * フローティングアクションボタンコンポーネント
 * @return {React.ReactElement}
 */
const FloatingActionButton: React.FC<FloatingActionButtonProps> = (
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  _props?: FloatingActionButtonProps
): React.ReactElement => {
  // ダイアログの開閉
  const [dialogActivated, setDialogActivated] = React.useState(false);

  return (
    <>
      <FormDialog dialogActivated={dialogActivated} onClose={() => setDialogActivated(false)} />
      <Fab sx={{ ...style }} size="medium" color="secondary" aria-label="add" onClick={() => setDialogActivated(true)}>
        <AddIcon />
      </Fab>
    </>
  );
};

export default FloatingActionButton;
