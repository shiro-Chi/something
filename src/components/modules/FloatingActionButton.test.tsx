import React from 'react';
import { render } from '@testing-library/react';
import FloatingActionButton from './FloatingActionButton';

describe('スナップショットテスト', () => {
  test('前回スナップショットと一致すること', () => {
    const sut = render(<FloatingActionButton />);
    expect(sut).toMatchSnapshot();
  });
});
