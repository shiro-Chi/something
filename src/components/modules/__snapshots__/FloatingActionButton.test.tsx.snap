// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`スナップショットテスト 前回スナップショットと一致すること 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-label="add"
        class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeMedium MuiFab-secondary MuiFab-root MuiFab-circular MuiFab-sizeMedium MuiFab-secondary css-1qr62f1-MuiButtonBase-root-MuiFab-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AddIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      aria-label="add"
      class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeMedium MuiFab-secondary MuiFab-root MuiFab-circular MuiFab-sizeMedium MuiFab-secondary css-1qr62f1-MuiButtonBase-root-MuiFab-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="AddIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
